src/Rules/Comparison/ImpossibleCheckTypeHelper.php

```diff
@@ -132,6 +132,56 @@ public function findSpecifiedType(
 							}
 						}
 					}
+				} elseif (
+					$functionName === 'array_key_exists'
+					&& count($node->args) >= 2
+				) {
+					$haystackType = $scope->getType($node->args[1]->value);
+					if ($haystackType instanceof MixedType) {
```

Comment:
isn't mixed also covered by the following IF and therefore this case here can be dropped?

Reply:
the same check is used for "in_array" here, so I think it's ok